mobile/views/TabBar/headerActions/NavigateToThreadDetails.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from 'react';
+import type { NavigationProps } from 'react-navigation';
+import { Button } from 'react-native';
+
+type Props = {
+  navigation: NavigationProps,
+};
+
+export default class NavigateToThreadDetails extends React.Component<Props> {
+  navigate = () => {
+    const { navigation } = this.props;
+    return navigation.navigate({
+      routeName: `ThreadDetail`,
+      key: `thread-detail-${navigation.state.key}`,
```

Comment:
I haven't seen this kind of custom key on `navigate` anywhere else, what do we need it for?

Reply:
The reason is that `react-navigation` uses the `key` to know what view it is looking at and where it exists in the stack; in this case the `ThreadDetail` is loading the same thread as the `Thread` view, which would both have `thread.id` as the key - this breaks navigation. Every key needs to be unique for `react-navigation` to know how to push or go back to a new view.

Reply:
This is the first instance of this pattern, but it will probably come up again in the future if we add more `Detail`-type screens for communities, channels, users, etc

Reply:
Wait what's the thread detail view compared to the Thread view?

Reply:
Tap into a thread, then tap 'details' in the top right